File: CodeForces-1560C.cs
/*
* Created by rrenamed
* Feel free to contribute explanations
* https://codeforces.com/problemset/submission/1560/127459661
*/

class Program
{
    static void Main()
    {
        int test_cases = Convert.ToInt32(Console.ReadLine());
        while (test_cases-- != 0)
        {
            int _case = Convert.ToInt32(Console.ReadLine());
            int sqrtL = (int)Math.Floor(Math.Sqrt(_case));
            int sqrtB = (int)Math.Ceiling(Math.Sqrt(_case));
            if (sqrtB * sqrtB - _case > _case - sqrtL * sqrtL)
                Console.WriteLine("{0} {1}", _case - sqrtL * sqrtL, sqrtB);
            else
                Console.WriteLine("{0} {1}", sqrtB, sqrtB * sqrtB - _case + 1);
        }
        //Console.ReadKey();
    }
}

File: CodeForces-1560C.cpp
/*
* Created by rrenamed
* Feel free to contribute explanations
*/

#include <iostream>
#include <math.h>

using namespace std;
int main() {
    int test_cases;
    cin >> test_cases;
    //Getting the number of testing cases
    while (test_cases--)
    {
        int _case;
	    cin >> _case;
        //Getting case
        int sqrtF = floor(sqrt(_case));
        //Getting floor of case
        int sqrtC = ceil(sqrt(_case));
        //Ceiling case
        if (sqrtC * sqrtC - _case > _case - sqrtF * sqrtF) //Checking what sqrt is closer to case
            cout << _case - sqrtF * sqrtF << " " << sqrtC << "\n";
        else
            cout << sqrtC << " " << sqrtC * sqrtC - _case + 1 << "\n";
    }
  return 0;
}
